app/src/main/java/com/sunlightlabs/congress/services/ProPublica.java

```diff
@@ -0,0 +1,173 @@
+
+        return baseUrl + path.toString() + query.toString();
+    }
+
+    public static String fetchJSON(String url) throws CongressException {
+        Log.d(Utils.TAG, "Pro Publica API: " + url);
+
+        // play nice with OkHttp
+        HttpManager.init();
+
+        HttpURLConnection connection;
+        URL theUrl;
+
+        try {
+            theUrl = new URL(url);
+            connection = (HttpURLConnection) theUrl.openConnection();
+        } catch(MalformedURLException e) {
+            throw new CongressException(e, "Bad URL: " + url);
+        } catch (IOException e) {
+            throw new CongressException(e, "Problem opening connection to " + url);
+        }
+
+        try {
+            // Identify ourselves as the Congress app
+            connection.setRequestProperty("User-Agent", userAgent);
+
+            // Supply the Pro Publica API key over a header
+            connection.setRequestProperty("X-API-Key", apiKey);
+
+            int status = connection.getResponseCode();
+            if (status == HttpURLConnection.HTTP_OK) {
+                // read input stream first to fetch response headers
+                InputStream in = connection.getInputStream();
+
+                // adapted from https://stackoverflow.com/a/2549222/16075
+                BufferedReader reader = new BufferedReader(new InputStreamReader(in));
+                StringBuilder total = new StringBuilder();
+                String line;
+                while ((line = reader.readLine()) != null) total.append(line);
+
+                return total.toString();
+
+            } else if (status == HttpURLConnection.HTTP_NOT_FOUND)
+                throw new CongressException.NotFound("404 Not Found from " + url);
+            else
+                throw new CongressException("Bad status code " + status+ " on fetching JSON from " + url);
+
+        } catch (IOException e) {
+            throw new CongressException(e, "Problem fetching JSON from " + url);
+        } finally {
+            connection.disconnect();
+        }
+    }
+
+    public static JSONObject firstResult(String url) throws CongressException {
+        JSONArray results = resultsFor(url);
+        if (results.length() > 0) {
+            try {
+                return (JSONObject) results.get(0);
+            } catch(JSONException e) {
+                throw new CongressException(e, "Error getting first result from " + url);
+            }
+        } else
+            return null;
+    }
+
+    public static JSONArray resultsFor(String url) throws CongressException {
+        String rawJSON = fetchJSON(url);
+        JSONArray results = null;
+        try {
+            JSONObject response = new JSONObject(rawJSON);
+
+            // First check that the Pro Publica API said 'OK'
+            String status = response.getString("status");
+            if (!status.equals("OK"))
+                throw new CongressException("Got a non-OK status from " + url + "\n\n" + rawJSON);
```

Comment:
@dwillis I'm currently just making sure that the `status` field is `"OK"` before proceeding to parse. I assume it'll be some non-`"OK"` value in any of the [error conditions](https://projects.propublica.org/api-docs/congress-api/error/).